dcm2niix_gear/utils/resolve.py

```diff
@@ -162,20 +162,31 @@ def retain_gear_outputs(
 
         # Move data files, if indicated
         for file in output_image_files:
-
-            if Path(sidecar).stem in Path(file).stem:
-
-                file_type = "".join(Path(file).suffixes)
-
-                if retain_nifti:
-                    if file_type in [".nii.gz", ".nii", ".bval", ".bvec"]:
-                        log.info(f"Moving {file} to output directory.")
-                        shutil.move(file, output_dir)
-
-                if output_nrrd:
-                    if file_type in [".raw.gz", ".nhdr", ".nrrd"]:
-                        log.info(f"Moving {file} to output directory.")
-                        shutil.move(file, output_dir)
+            # Split sidecar into the "root" name by removing .json suffix
+            stem = sidecar.split('.json')[0]
+            # Split image name by "root" name from sidecar
+            substr = file.split(stem)
+            # If the "root" name of sidecar is not a substring of image file
+            #   We don't care about this image file.
+            if len(substr) < 2:
+                continue
+            # Get "root" and what's "left" over after splitting of "root" name
+            #   from sidecar
+            substr = root, left
```

Comment:
Does this work? I thought the RHS and LHS would be swapped here?

Reply:
Shoudl be remove. And root, left are undefined and that is unused.